presto-memory-context/src/main/java/com/facebook/presto/memory/context/AggregatedMemoryContext.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.memory.context;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.util.OptionalLong;
+
+import static com.facebook.presto.memory.context.MemoryReservationHandler.NOT_BLOCKED;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+@ThreadSafe
+public class AggregatedMemoryContext
```

Comment:
Rename this to `Aggregated*Resource*Context`, so that it can be used to track spilled data size across operators too. Currently we use very similar classes to track spilled data size (which has precisely same tracking pattern as allocated memory) and the mechanisms should be unified.

See also https://github.com/prestodb/presto/pull/9690/commits/6f41dc365cbf03c99818d46cc0044aa722a4812a#diff-0ce8d01e60a53ad1081516a08cddcc7c where I picked `Resource` as a common name for memory and spilled data usage.
  
 (Also, make it `final`)